docs/DeveloperGuide.md

```diff
@@ -6,8 +6,248 @@
+
+Step 6:I f there are is an item with a matching name found, we would then call the `UI` object and execute the `UI`'s `printItemAlreadyInTheListErrorMessage` method which would print a message informing the user that the item the user wants to add is already found within the item list and hence nothing would be added. Step 7 and 8 are skipped.
+
+Step 7: If there are no items with a matching name found, we would add the `Item` object to its ArrayList of `Item` objects called listOfItems. The `AddItemCommand` then call the `UI` and execute the `UI`'s `printAddItemAcknowledgementMessage` method which would print an acknowledgement message to the user informing him that the item has been added into the item list.
+
+Step 8: The `ItemListFileManager` object would be called and we would pass the item list to its `writeItemListToFile` method to write the updated item list into the file ListFolder/ItemList.txt.
+
+## Implementation
+
+#### Assign a housekeeper to a room
+The assignment of housekeeper to a room is facilitated by `AssignHousekeeperCommand`. This command objects uses `AssignmentMap`, which keeps a `HashMap`, to help us map a `Room` object to a `Housekeeper` object. 
+
+``
+Map<Room, Housekeeper> map = new HashMap<>();
+``
+
+The following steps exhibit how `AssignHousekeeperCommand` 
+and `AssignmentMap` fit together in the structure and how the mechanism for adding an assinment behaves at each step.
+
+__Step 1.__ The user launches the application. In the `Duke` class, an empty instance of the `AssignmentMap` class,
+called `assignmentMap`, is created.
+
+![Step 1](team/aiman_assignment/step1.png)
+
+__Step 2.__ The user types the command `Assign Susan ## 301`. In the `Duke` class, a `Command` object
+is created by invoking the `CommandParser` class's constructor on the user input. The details of this
+step are further described below.
+
+![Step 2](team/aiman_assignment/step2.png)
+
+__Step 3.__ The `CommandParser` class replaces the `Assign Susan ## 301` in the user input with an empty string,
+leaving just `Susan ## 301`. Then, an instance of `AssignHousekeeperCommand` is created which extends the class `Command`.
+
+![Step 3](team/aiman_assignment/step3.png)
+
+__Step 4.__ The `assignHousekeeperCommand` object parses the name of the housekeeper and the ID of the room and send both to the the `assignmentMap`.
+
+![Step 4](team/aiman_assignment/step4.png)
+
+__Step 5.__ The `assignmentMap` looks for the appropriate `Room` object for `301` in the `RoomList` and then looks for the appropriate `Housekeeper` object for `Susan`. It then adds both to the hashmap contained inside itself.
+
+![Step 5](team/aiman_assignment/step5.png)
+
+__Interaction__ To understand the interaction between objects of these classes in order to actually add an entry into the hash map inside an `AssignmentMap`, refer to the sequence diagram below.
+
+![Step Sequence Diagram](team/aiman_assignment/sequence.png)
```

Comment:
A few issues with the sequence diagram:
1. Return arrows should simply contain the object it returns, `return` can be omitted.
2. If nothing is returned, it can simply be a blank dotted arrow pointing back to the caller.
3. Syntax of content for return arrows should be as follows: variableName: ClassName

Consider making the following changes for better clarity:
1. Calling arrow for `AssignmentMap#addAssignment` requires a closing bracket in the text
2. Return arrow for `RoomList#getRoomList` can simply be labelled `rooms: ArrayList<Room>`
3. Return arrow for `HousekeeperList#getHousekeeperList` can simply be labelled `hs: ArrayList<Housekeeper>`
4. Return arrow for `AssignmentMap#houseKeeperExists` should be `found: boolean` for type `boolean`. Additionally, the set of arrows for `AssignmentMap#houseKeeperExists` should be enlarged as it is very hard to read at this size.
5. Return arrows from `:AssignmentMap` to `;AssignHousekeeperCommand` and onwards can simply be blank.